### PR TITLE
Fix bug on pull requests when transfer head repository

### LINKED ISSUE #8563: Transfer a repository will result in pull requests from the same repository broken
There is a duplicated column named `HeadUserName` on `PullRequest`. But if `HeadRepo` transfered to a new place, we haven't change the pull requests' `HeadUserName`.

There are two method to resolve the problem. One is update all `PullRequest`'s `HeadUserName` when transfered repository.

The second is to remove the column `HeadUserName` and get it from `Repository` and `User`.

Which one is better?

### PR DESCRIPTION
**Pull request by lunny**
Fix #8563

### CODE REVIEW

**Review by guillep2k — APPROVED** [review]
Code looks good to me. I'm still not sure what was the actual use of that column.

**Comment by lafriks** [comment]
We can not migrate this to 1.9 because of migrations

File: models/pull.go [hunk 1]
```diff
@@ -113,6 +121,24 @@ func (pr *PullRequest) LoadBaseRepo() error {
...
+			pr.HeadRepo = pr.BaseRepo
+			return nil
+		}
+		var repo Repository
+		if has, err := x.ID(pr.HeadRepoID).Get(&repo); err != nil {
+			return err
+		} else if !has {
+			return ErrRepoNotExist{ID: pr.BaseRepoID}
+		}
+		pr.HeadRepo = &repo
```
[the remark this answers is not in the document]
Comment (lunny) [reply models/pull.go:141]: Done. Not here but in `LoadBaseRepo`.

File: routers/repo/pull.go [hunk 2]
```diff
@@ -531,8 +531,8 @@ func ViewPullFiles(ctx *context.Context) {
 		endCommitID = headCommitID
 		gitRepo = headGitRepo
 
-		headTarget = path.Join(pull.HeadUserName, pull.HeadRepo.Name)
-		ctx.Data["Username"] = pull.HeadUserName
+		headTarget = path.Join(pull.MustHeadUserName(), pull.HeadRepo.Name)
```
[the remark this answers is not in the document]
Comment (lunny) [reply routers/repo/pull.go:534]: @guillep2k No. `pull.HeadRepo.RepoPath()` will return repo's location on file system. But here we need a `headTarget`.

**Review by lafriks — APPROVED** [review]

**Pull request merged**